torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1172,7 +1156,9 @@ def is_number(s: str):
                 # ArrayRefTensors. The code flowing into here uses `0` for nullptr,
                 # which convert_arrayref_tensor_to_tensor would blindly coerce to int,
                 # so just avoid wrapping integers.
-                if not is_number(piece):
+                # Name matching is to find tensor is hacky, but fixing all the
+                # ArrayRefTensor issues is not a priority for now.
+                if isinstance(piece, str) and piece.startswith(("buf", "arg")):
```

Comment:
I know this is hacky, but the previous `is_number` detection was hacky anyway. A proper way to do it is to pass in type information, but as I mentioned in the comment, supporting ArrayRef is not a priority now.